src/System.Text.JsonLab/System/Text/Json/JsonWriterUtf16.cs

```diff
@@ -543,12 +543,37 @@ public void WriteValue(Guid value)
         /// </summary>
         public void WriteNull()
         {
-            //TODO: Optimize, just like WriteValue(long value)
-            WriteItemSeperatorUtf16();
-            _firstItem = false;
-            WriteSpacingUtf16();
+            int charsNeeded = (_firstItem ? 0 : 1) + (_prettyPrint ? 2 + (_indent + 1) * 2 : 0);
```

Comment:
The calculation of `charsNeeded` and `bytesNeeded` should be in a checked block without some other mechanism guaranteeing that they can't integer overflow.

Reply:
Do you think that an _indent value of 2^30 is realistic for writing some json payload? Only then would a calculation like this overflow.

Reply:
It appear to me that none of the calculate buffer size logic are checked - is that really an issue?